src/lib/rng/rng.h

```diff
@@ -40,18 +40,28 @@ class BOTAN_PUBLIC_API(2,0) RandomNumberGenerator
       RandomNumberGenerator(const RandomNumberGenerator& rng) = delete;
       RandomNumberGenerator& operator=(const RandomNumberGenerator& rng) = delete;
 
+   protected:
```

Comment:
That's the main (functional) method that needs to be overridden by subclasses and I wanted it to be "prominent" in the interface. No hard feelings about moving it down though.

Reply:
I would tend to say it is more important to write it in a way that is useful for users of a `RandomNumberGenerator`, instead of people wanting to create their own derived subclass.

But I'm also OK with leaving as is, hopefully the documentation is good enough that users don't need to read the header to understand how to use a RNG in Botan.